azure/scope/machinepool.go

```diff
@@ -436,16 +453,17 @@ func (m *MachinePoolScope) applyAzureMachinePoolMachines(ctx context.Context) er
 		return nil
 	}
 
-	// select machines to delete to lower the replica count
+	// Select Machines to delete to lower the replica count
 	toDelete, err := deleteSelector.SelectMachinesToDelete(ctx, m.DesiredReplicas(), existingMachinesByProviderID)
 	if err != nil {
 		return errors.Wrap(err, "failed selecting AzureMachinePoolMachine(s) to delete")
 	}
 
-	for _, machine := range toDelete {
-		machine := machine
-		log.Info("deleting selected AzureMachinePoolMachine", "providerID", machine.Spec.ProviderID)
-		if err := m.client.Delete(ctx, &machine); err != nil {
+	// Delete MachinePool Machines as a part of scaling down
+	for i := range toDelete {
+		ampm := toDelete[i]
+		log.Info("deleting selected AzureMachinePoolMachine", "providerID", ampm.Spec.ProviderID)
```

Comment:
```suggestion
		log.V(2).Info("deleting selected AzureMachinePoolMachine", "providerID", ampm.Spec.ProviderID)
```

Maybe? We seem to prefer `V(2)` in most other create/delete resource code.

Reply:
That log is from the existing code so I just kept it at the same level. IIRC log level 0 is for actions while level 2 is more for information.